### PR TITLE
FIX-#5171: Allow xgboost >= 1.7.0.

### LINKED ISSUE #5171: TEST: CI: test_xgboost 1.7 on test-ubuntu[ray] is getting stuck
Happening on two unrelated PRs: 

1) https://github.com/modin-project/modin/actions/runs/3366993128/jobs/5584094005
2) https://github.com/modin-project/modin/actions/runs/3365902084/jobs/5583379988

Lasting over 2 hours while normally it's probably < 10 minutes.

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5171 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by pyrito — APPROVED** [review]

File: environment-dev.yml [hunk 1]
```diff
@@ -40,9 +40,9 @@ dependencies:
   - pandas-stubs
   - fastparquet
   - pip:
-      # TODO(https://github.com/modin-project/modin/issues/5171): uncap xgboost
-      # when we can run tests on 1.7.0 without getting stuck.
-      - xgboost>=1.6.0,<1.7.0
```
Comment (anmyachev) [inline environment-dev.yml:45]: I think we should not remove the lower bound.

File: requirements-dev.txt [hunk 2]
```diff
@@ -29,9 +29,9 @@ cloudpickle
 rpyc==4.1.5
 scikit-learn
 git+https://github.com/airspeed-velocity/asv.git@ef016e233cb9a0b19d517135104f49e0a3c380e9
-# TODO(https://github.com/modin-project/modin/issues/5171): uncap xgboost
-# when we can run tests on 1.7.0 without getting stuck.
-xgboost>=1.6.0,<1.7.0
+# TODO(https://github.com/modin-project/modin/issues/5194): Uncap xgboost
+# when we use collective instead of rabit.
+xgboost>=1.6.0,<2.0.0
```
Comment (vnlitvinov) [inline requirements-dev.txt:34]: This feels wrong, it should probably be `>=1.7.1`, as 1.7.0 is broken for us and a fix is made in 1.7.1...

File: environment-dev.yml (hunk 1 above)
Comment (vnlitvinov) [reply environment-dev.yml:45]: agreed, let's not lose pinnings

**Review by vnlitvinov — CHANGES_REQUESTED** [review]

File: requirements-dev.txt (hunk 2 above)
Comment (mvashishtha) [reply requirements-dev.txt]: Now using xgboost>=1.7.1,<2.0.0 everywhere

File: environment-dev.yml (hunk 1 above)
Comment (mvashishtha) [reply environment-dev.yml:45]: Now using xgboost>=1.7.1,<2.0.0 everywhere

File: requirements/requirements-no-engine.yml [hunk 3]
```diff
@@ -30,9 +30,9 @@ dependencies:
   - cloudpickle
   - boto3
   - pip:
-      # TODO(https://github.com/modin-project/modin/issues/5171): uncap xgboost
-      # when we can run tests on 1.7.0 without getting stuck.
-      - xgboost>=1.6.0,<1.7.0
+      # TODO(https://github.com/modin-project/modin/issues/5194): Uncap xgboost
+      # when we use collective instead of rabit.
+      - xgboost>=1.7.1,<2.0.0
```
Comment (anmyachev) [inline requirements/requirements-no-engine.yml:35]: @vnlitvinov should we add the same pins for xgboost in `env_hdk.yml`?

Comment (vnlitvinov) [reply requirements/requirements-no-engine.yml:35]: I have no idea as I don't have anything to do with HDK anymore, but I would like to have our requirements be as consistent as possible.

**Review by vnlitvinov — COMMENTED** [review]
@mvashishtha let's fix all `.yml` and `.txt` files, I think `env_hdk.yml` (and maybe `requirements-doc.txt`, not sure about that tho as it doesn't pin any version in it) is still missing.

File: requirements/requirements-no-engine.yml (hunk 3 above)
Comment (mvashishtha) [reply requirements/requirements-no-engine.yml:35]: I added the same requirement for hdk

**Comment by mvashishtha** [comment]
@vnlitvinov I've added env_hdk and requirements-doc pins even though those don't pin xgboost already. Also we seem to support xgboost on ray only and it's not clear to me that hdk even uses xgboost. Anyway, everything that requires xgboost now specifies the same versions.

**Review by anmyachev — APPROVED** [review]

**Comment by anmyachev** [comment]
@vnlitvinov up to you

**Pull request merged**